src/app/core/dashboard/dashboard-list-widget/dashboard-list-widget.component.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  AfterViewInit,
+  Component,
+  ContentChild,
+  Input,
+  OnChanges,
+  SimpleChanges,
+  ViewChild,
+} from "@angular/core";
+import { IconName } from "@fortawesome/fontawesome-svg-core";
+import { DashboardTheme } from "../dashboard-widget/dashboard-widget.component";
+import { MatTable, MatTableDataSource } from "@angular/material/table";
+import { MatPaginator } from "@angular/material/paginator";
+import { BehaviorSubject } from "rxjs";
+import { EntityMapperService } from "../../entity/entity-mapper.service";
+import { filter, map } from "rxjs/operators";
+import { applyUpdate } from "../../entity/model/entity-update";
+
+/**
+ * Base dashboard widget to build widgets that display a number of entries as a table.
+ *
+ * This widget automatically handles loading animation and pagination.
+ * Define a mat-table with your desired layout for displaying the entries through content-projection
+ * and provide the data array as an input using async pipe.
+ *
+ * <app-dashboard-list-widget [entries]="notes | async">
+ *   <table mat-table>
+ *     <!-- define row layout here-->
+ *   </table>
+ * </app-dashboard-list-widget>
+ *
+ * To highlight a row with a colored bar at the front, apply `class="row-indicator"` to your first <td>
+ *   and set the css variable to the desired color, e.g. `[ngStyle]="{'--row-indicator-color': row.getColor?.()}"`
+ */
+@Component({
+  selector: "app-dashboard-list-widget",
+  templateUrl: "./dashboard-list-widget.component.html",
+  styleUrls: ["./dashboard-list-widget.component.scss"],
+})
+export class DashboardListWidgetComponent<E>
+  implements OnChanges, AfterViewInit
+{
+  @Input() subtitle: string;
+  @Input() icon: IconName;
+  @Input() theme: DashboardTheme;
+  @Input() title: string | number;
+  /** optional tooltip to explain detailed meaning of this widget / statistic */
+  @Input() explanation: string;
+  @Input() headline: string;
+
+  /**
+   * array of items to be displayed in paginated widget table
+   * you can use an observable with async pipe here to let the component automatically handle "loading" indicator.
+   *
+   * Alternatively define an entityType and dataPipe to let the component load data itself.
+   */
+  @Input() entries: E[];
+
+  /**
+   * entity type to be loaded for displaying entries
+   * (filter or transform entities using the dataPipe input).
+   *
+   * If you define an entityType, the "entries" input is ignored in favor of directly loading data here
+   */
+  @Input() entityType: string;
+
+  /**
+   * Pipe to map, filter or sort the loaded data
+   */
+  @Input() dataMapper: (data: E[]) => E[];
+
+  isLoading: boolean = true;
+  data = new BehaviorSubject<E[]>(undefined);
```

Comment:
Could we also use a Subject here? This way you wouldn't have to filter out the undefined values.

Reply:
good point, not sure why I don't like Subject :sweat_smile: 

I am leaving the filter for undefined in addition though because the `entries` input might deliver undefined potentially.

Reply:
hmm, actually I am using `data.value` for the receiveUpdates - applyUpdate (that's even the main reason of having this.data). And that is only supported by BehaviorSubject.